### PR TITLE
Part 5

### PR DESCRIPTION
**Pull request by minhng01**
Completed part 5

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -120,6 +122,22 @@ OvenStove::FoodItem::FoodItem() : type("Poultry"),
     std::cout << "----------constructing FoodItem----------" << "\n";
 }
 
+void OvenStove::FoodItem::storeInFridge(int threshold, int expirationDuration){
+    expirationDayRemain = expirationDuration;
+    
+    while (expirationDayRemain > threshold)
+    {
+        std::cout << "There are " << expirationDayRemain << " days till expired. Keep in fridge.\n";
+        expirationDayRemain--;
```
Comment (matkatmusic) [inline main.cpp]: - Curly braces go on their own line with no other code
- use `--i`, not `i--`
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -175,6 +193,22 @@ void OvenStove::FoodItem::cut(int numberOfPortion, std::string tool)
...
 }
 
+void OvenStove::cookFoodOnBurner(int threshold, int totalBurner)
+{
+    numberOfBurner = totalBurner;
+    
+    while (numberOfBurner > threshold)
+    {
+        std::cout << "Cook food on burner no." << numberOfBurner << std::endl;
+        numberOfBurner--;
```
Comment (matkatmusic) [inline main.cpp]: use `--i`, not `i--`
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -238,6 +273,29 @@ Laptop::Laptop() : modelName("Thinkpad"),
...
+{
+    storageSize = availableDiskStorage;
+    
+    std::cout << "There is " << availableDiskStorage << "GB storage in total.\n";
+    std::cout << "How many " << eachFileSize << "GB files that can be stored given that " 
+                << storageForOS << " GB should be left for system operation and only "
+                << maxNumOfFile << " maximum files that can be stored?\n";
+    
+    int numberOfFile = 0;
+    for (int i = 1; i <= ((availableDiskStorage - storageForOS) / eachFileSize); i++)
```
Comment (matkatmusic) [inline main.cpp]: use `++i`, not `i++`
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -300,6 +359,24 @@ FruitTree::FruitTree() : name("Apple Tree"),
     std::cout << "----------constructing FruitTree----------" << "\n";
 }
 
+double FruitTree::treeGrowTrack(int currentAge, int lifeSpan, double maxSize)
+{
+    std::cout << "Current age: " << currentAge << "\n";
+    std::cout << "Current barkThicknessDiameter: " << barkThicknessDiameter << "\n";
+    std::cout << "Max life span (year): " << lifeSpan << "\n";
+
+    for (int i = currentAge; i <= lifeSpan; i++)
```
Comment (matkatmusic) [inline main.cpp]: use `++i`, not `i++`
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -411,6 +523,37 @@ void Camera::Lens::getFocus(double distanceToObject)
...
+        if (currentWhiteBalanceMode < target)
+        {   
+            std::cout << "Current white balance mode: " << currentWhiteBalanceMode << std::endl;
+            currentWhiteBalanceMode++;
+        }
+        else if (currentWhiteBalanceMode > target)
+        {
+            std::cout << "Current white balance mode: " << currentWhiteBalanceMode << std::endl;
+            currentWhiteBalanceMode--;
+        }
```
Comment (matkatmusic) [inline main.cpp]: Everywhere that you have `i++`, `i--`, `i -= 1;` or `i += 1;`, you should use `pre-increment/decrement`. 
You are currently doing `post-increment/decrement`.  
you can think of post- and pre- increment like this:
```
int preIncrement(int& i)
{
    i += 1;
    return i; //both the returned value AND i have changed
}
​
int postIncrement(int& i)
{
    int oldVal = i;
    i += 1;
    return oldVal; //i has changed but the value being returned does not reflect this change.
}
```
more information can be found here explaining why `pre-increment/decrement` is preferred over `post-increment/decrement`: https://en.cppreference.com/w/cpp/language/operator_incdec
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -470,6 +616,20 @@ Cooktop::Cooktop() : numberOfBurner(4),
     std::cout << "----------constructing Cooktop----------" << "\n";
 }
 
+void Cooktop::turnOnBurner(int target)
+{
+    std::cout << "Number of burners in use: " << numberOfBurnerInUse << std::endl;
+    for (int i = numberOfBurnerInUse; i <= target; i++)
+    {
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-increment

File: main.cpp [hunk 7]
```diff
@@ -520,6 +681,16 @@ Oven::Oven() : numberOfRack(2),
     std::cout << "----------constructing Oven----------" << "\n";
 }
 
+void Oven::turnOnLight(int target)
+{
+    int numberOfLightOn = 0;
+    while (numberOfLightOn <= target && numberOfLightOn <= numberOfLightBulb)
+    {
+        std::cout << "Turning " << numberOfLightOn << " oven light bulbs on.\n";
+        numberOfLightOn++;
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-increment

File: main.cpp [hunk 8]
```diff
@@ -572,6 +745,20 @@ Control::Control()
     std::cout << "----------constructing Control----------" << "\n";
 }
 
+void Control::turnBurnerKnob(int target)
+{
+    std::cout << "Number of knobs turned: " << numberOfTurnedBurnerKnob << std::endl;
+    for (int i = numberOfTurnedBurnerKnob; i <= target; i++)
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-increment

File: main.cpp [hunk 9]
```diff
@@ -613,6 +801,16 @@ RangeHood::RangeHood()
     std::cout << "----------constructing RangeHood----------" << "\n";
 }
 
+void RangeHood::turnOnFan(int targetFanLevel)
+{
+    int currentFanLevel = 2;
+    while (currentFanLevel <= targetFanLevel && currentFanLevel <= numberOfFanStrengthLevel)
+    {
+        std::cout << "Adjusted current fan level: " << currentFanLevel << ".\n";
+        currentFanLevel++;
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-increment

File: main.cpp [hunk 10]
```diff
@@ -660,6 +859,22 @@ UnderStoveStorage::UnderStoveStorage()
     std::cout << "----------constructing UnderStoveStorage----------" << "\n";
 }
 
+void UnderStoveStorage::putPanIn(int numberOfPan)
+{
+    for (int i = 1; i <= numberOfPan; i++)
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-increment

File: main.cpp [hunk 11]
```diff
@@ -660,6 +859,22 @@ UnderStoveStorage::UnderStoveStorage()
...
+    {
+        if (capacity <= 0)
+        {
+            std::cout << "Storage is FULL!\n";
+            capacity = 0;
+            return;
+        }
+
+        std::cout << "Storing " << i << " pans in storage.\n";
+        capacity--;
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-decrement

File: main.cpp [hunk 12]
```diff
@@ -702,6 +918,18 @@ KitchenRange::KitchenRange()
...
 
+void KitchenRange::cookTime(int cookTimeLeft, int notifyThreshold)
+{
+    while (cookTimeLeft >= 0)
+    {
+        if (cookTimeLeft == notifyThreshold)
+            std::cout << "Watch out: " << cookTimeLeft << " hours left on cooking timer.\n";
+
+        std::cout << cookTimeLeft << " hours left on cooking timer.\n";
+        cookTimeLeft--;
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise, use pre-decrement

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 5 above)
Comment (minhng01) [reply main.cpp]: Thank you for this review and the info!

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**